qiskit/circuit/library/arithmetic/adder/ripple_carry_adder.py

```diff
@@ -0,0 +1,120 @@
+                     │      ││      ││      │     │      ││      ││      │
+          input_a_2: ┤      ├┤  MAJ ├┤0     ├──■──┤0     ├┤  UMA ├┤      ├
+                     │      ││      ││      │  │  │      ││      ││      │
+          input_b_0: ┤1 MAJ ├┤      ├┤  MAJ ├──┼──┤  UMA ├┤      ├┤1 UMA ├
+                     │      ││      ││      │  │  │      ││      ││      │
+          input_b_1: ┤      ├┤1     ├┤      ├──┼──┤      ├┤1     ├┤      ├
+                     │      │└──────┘│      │  │  │      │└──────┘│      │
+          input_b_2: ┤      ├────────┤1     ├──┼──┤1     ├────────┤      ├
+                     │      │        └──────┘  │  └──────┘        │      │
+         carry_in_0: ┤2     ├──────────────────┼──────────────────┤2     ├
+                     └──────┘                ┌─┴─┐                └──────┘
+        carry_out_0: ────────────────────────┤ X ├────────────────────────
+                                             └───┘
+
+    Here *MAJ* and *UMA* gates correspond to the gates introduced in [1]. Note that
+    in this implementation the input register qubits are ordered as all qubits from
+    the first input register, followed by all qubits from the second input register.
+    This is different ordering as compared to Figure 4 in [1], which leads to a different
+    drawing of the circuit.
+
+    **References:**
+
+    [1] Cuccaro et al., A new quantum ripple-carry addition circuit, 2004.
+    `arXiv:quant-ph/0410184 <https://arxiv.org/pdf/quant-ph/0410184.pdf>`_
+
+    [2] Vedral et al., Quantum Networks for Elementary Arithmetic Operations, 1995.
+    `arXiv:quant-ph/9511018 <https://arxiv.org/pdf/quant-ph/9511018.pdf>`_
+    """
+
+    def __init__(self,
+                 num_state_qubits: int,
+                 name: str = 'RippleCarryAdder'
+                 ) -> None:
+        r"""
+        Args:
+            num_state_qubits: The number of qubits in either input register for
+                state :math:`|a\rangle` or :math:`|b\rangle`. The two input
+                registers must have the same number of qubits.
+            name: The name of the circuit object.
+        Raises:
+            ValueError: If ``num_state_qubits`` is lower than 1.
+        """
+        if num_state_qubits < 1:
+            raise ValueError('The number of qubits must be at least 1.')
+
+        qr_a = QuantumRegister(num_state_qubits, name='input_a')
+        qr_b = QuantumRegister(num_state_qubits, name='input_b')
+        qr_z = QuantumRegister(1, name='carry_out')
+        qr_c = AncillaRegister(1, name='carry_in')
+
+        # initialize quantum circuit with register list
+        super().__init__(qr_a, qr_b, qr_z, qr_c, name=name)
+
+        # build carry circuit for majority of 3 bits in-place
+        # corresponds to MAJ gate in [1]
+        qc_maj = QuantumCircuit(3, name='MAJ')
+        qc_maj.cnot(0, 1)
+        qc_maj.cnot(0, 2)
+        qc_maj.toffoli(2, 1, 0)
```

Comment:
Could you use `cx` and `ccx` instead? They are mostly used and I'm not sure we'll be keeping `toffoli` indefinitely 🙂 
```suggestion
        qc_maj.cx(0, 1)
        qc_maj.cx(0, 2)
        qc_maj.ccx(2, 1, 0)
```

Reply:
I have pushed a commit with the fix, will close the conversation as resolved, since I think it's pretty clear :)